lib/cli_common/cli_common/taskcluster.py

```diff
@@ -182,3 +187,42 @@ def get_hook_artifact(hook_group_id, hook_id, artifact_name, client_id=None,
 
     # Load artifact from task run
     return queue.getArtifact(task_id, run_id, artifact_name)
+
+
+def create_blob_artifact(queue_service, task_id, run_id, path, content, content_type, ttl):
+    '''
+    Manually create and upload a blob artifact to use a specific content type
+    '''
+    assert isinstance(content, str)
+    assert isinstance(ttl, datetime.timedelta)
+
+    # Create artifact on Taskcluster
+    sha256 = hashlib.sha256(content.encode('utf-8')).hexdigest()
+    resp = queue_service.createArtifact(
+        task_id,
+        run_id,
+        path,
+        {
+            'storageType': 'blob',
+            'expires': (datetime.datetime.utcnow() + ttl).strftime(TASKCLUSTER_DATE_FORMAT),
+            'contentType': content_type,
+            'contentSha256': sha256,
+            'contentLength': len(content),
+        }
+    )
+    assert resp['storageType'] == 'blob', 'Not a blob storage'
+    assert len(resp['requests']) == 1, 'Should only get one request'
+    request = resp['requests'][0]
+    assert request['method'] == 'PUT', 'Should get a PUT request'
+
+    # Push the artifact on storage service
+    push = requests.put(
+        url=request['url'],
+        headers=request['headers'],
+        data=content,
+    )
+    push.raise_for_status()
+
+    # Build the absolute url
+    # TODO: use f-string once all project are on python 3.6
```

Comment:
Which project is not using 3.6 yet?

Reply:
`mapper/api` was updated during this PR in huge #1730